### DISCUSSION TITLE
Is it possible to limit casadi's calculation time?

### DISCUSSION THREAD

**Discussion by PauloNTNU** [do-mpc/do-mpc#349] [Q&A] [answered]
Hi,

There's times casadi is taking too long with it's calculation and I would like to limit it, so it's forced to go into next step with "bad calculation" instead of taking too long.

**Comment by gioperin** [comment, marked as answer]
Hi @PauloNTNU,

you can set ipopt parameters when defining the problem. Specifically, you might be looking for reducing the maximum number of iteration (the default is 3000) before stopping. You can do it in this way (e.g., set to 500) in the `build_mpc` method:

`ipopt_params = {'ipopt.max_iter': 500}`
 `mpc.set_param(nlpsol_opts=ipopt_params)`

Best,

Giovanni

**Comment by 4flixt** [comment]
> Hi @PauloNTNU,
> 
> you can set ipopt parameters when defining the problem. Specifically, you might be looking for reducing the maximum number of iteration (the default is 3000) before stopping. You can do it in this way (e.g., set to 500) in the `build_mpc` method:
> 
> `ipopt_params = {'ipopt.max_iter': 500}` `mpc.set_param(nlpsol_opts=ipopt_params)`
> 
> Best,
> 
> Giovanni

Exactly this :)
